runtime/cudaq/matrix_op.h

```diff
@@ -176,12 +228,42 @@ class matrix_handler : public operator_handler {
 
   // predefined operators
 
+  /// @brief Constructs a product operator representing a number operator for
+  /// the given degree.
+  /// @param degree : The degree or power to which the number operator is
```

Comment:
"or power"? No, degree has nothing to do with power. The number operator acts on the given degree.